OpenRA.Platforms.Default/Shader.cs

```diff
@@ -134,14 +141,27 @@ public void PrepareRender()
 		{
 			VerifyThreadAffinity();
 			OpenGL.glUseProgram(program);
+			OpenGL.CheckGLError();
 
 			// bind the textures
 			foreach (var kv in textures)
 			{
+				var id = ((ITextureInternal)kv.Value).ID;
 				OpenGL.glActiveTexture(OpenGL.GL_TEXTURE0 + kv.Key);
-				OpenGL.glBindTexture(OpenGL.GL_TEXTURE_2D, ((ITextureInternal)kv.Value).ID);
+				OpenGL.glBindTexture(OpenGL.GL_TEXTURE_2D, id);
+
+				// HACK: Ignore the GL_INVALID_OPERATION that is thrown when starting a match from a lobby only with the threaded renderer enabled
```

Comment:
I really don't understand what is going on here - its fine when the render thread is disabled.
Any thoughts @RoosterDragon or @chrisforbes?

Reply:
`glDebugMessageCallback` logging showed that my checks for invalid textures above wasn't good enough, and we were still trying to bind a deleted texture here.

Rewrote the check to drop the manual state tracking and use `glIsTexture` instead.